src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,51 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationException("User can't be null");
+        }
+
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+
+        if (storageDao.get(user.getLogin()) == null) {
+            return storageDao.add(user);
+        }
+        throw new ValidationException("User already exist in database!");
+    }
+
+    private void validateLogin(String login) {
+        if (login == null) {
+            throw new ValidationException("Login can't be null");
+        } else if (login.length() < MIN_LENGTH) {
+            throw new ValidationException("Minimum allowed login length is " + MIN_LENGTH);
+        }
```

Comment:
Not sure in this case, but Vadym once wrote that better to avoid "if else" construction
https://medium.com/swlh/return-early-pattern-3d18a41bba8
```suggestion
        if (login == null) {
            throw new ValidationException("Login can't be null");
        }
        if (login.length() < MIN_LENGTH) {
            throw new ValidationException("Minimum allowed login length is " + MIN_LENGTH);
        }
```

Reply:
I read the article and have to agree - thanks for the good advice))